[DCS] Change Capacity from int to float64

Aloento opened the pull request:
## Summary of the Pull Request

Change Capacity from int to float32

## PR Checklist

* [x] Refers to: https://github.com/opentelekomcloud/terraform-provider-opentelekomcloud/issues/1839
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.
* [x] Release notes added.

## Acceptance Steps Performed

```
=== RUN   TestAccDcsInstancesV1_basic
=== PAUSE TestAccDcsInstancesV1_basic
=== CONT  TestAccDcsInstancesV1_basic
--- PASS: TestAccDcsInstancesV1_basic (107.52s)

PASS
```

Aloento committed:
[DCS] Change Capacity from int to float32

artem-lifshits reviewed (changes requested):
Tests are failing for me:
=== RUN   TestAccDcsInstancesV1_basic
=== PAUSE TestAccDcsInstancesV1_basic
=== CONT  TestAccDcsInstancesV1_basic
=== CONT  TestAccDcsInstancesV1_basic
    resource_opentelekomcloud_dcs_instance_v1_test.go:24: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
        -/+ destroy and then create replacement
        
        Terraform will perform the following actions:
        
          # opentelekomcloud_dcs_instance_v1.instance_1 must be replaced
        -/+ resource "opentelekomcloud_dcs_instance_v1" "instance_1" {
              + access_user         = (known after apply)
              ~ capacity            = 0 -> 0.125 # forces replacement
              ~ created_at          = "2022-07-29T11:09:12.225Z" -> (known after apply)
              + description         = (known after apply)
              ~ id                  = "3efae1b9-3b30-4d5c-91db-f2a86cf2ede5" -> (known after apply)
              + internal_version    = (known after apply)
              ~ ip                  = "192.168.1.229" -> (known after apply)
              ~ maintain_begin      = "02:00:00" -> (known after apply)
              ~ maintain_end        = "06:00:00" -> (known after apply)
              ~ max_memory          = 128 -> (known after apply)
                name                = "dcs_instance_0apw0"
              + order_id            = (known after apply)
              ~ port                = 6379 -> (known after apply)
              ~ resource_spec_code  = "redis.ha.xu1.tiny.r2.128" -> (known after apply)
              + security_group_id   = "738517ba-a383-4653-a5c9-4d225ee5736e"
              + security_group_name = (known after apply)
              ~ status              = "RUNNING" -> (known after apply)
              ~ subnet_name         = "subnet-testing" -> (known after apply)
              ~ used_memory         = 2 -> (known after apply)
              ~ user_id             = "087482cbabbc4d58890d2cfad91afd29" -> (known after apply)
              ~ user_name           = "alifshit" -> (known after apply)
              ~ vpc_name            = "vpc-testing" -> (known after apply)
                # (7 unchanged attributes hidden)
        
        
                # (2 unchanged blocks hidden)
            }
        
        Plan: 1 to add, 0 to change, 1 to destroy.
--- FAIL: TestAccDcsInstancesV1_basic (84.44s)

@anton-sidelnikov @vladimirvshivkov can you also check? Thanks

Aloento committed:
change Single to redis.single.xu1.tiny.128

vladimirvshivkov commented:
Fails for me too.
But I cant even create dcs in web console, I guess there is some quota issue.

Aloento committed:
update sdk

Aloento committed:
parse CapacityMinor

Aloento committed:
update (#1861)

* [ELB] Client from ctx in `resources/lb_v2` (#1852)

[ELB] Client from ctx in `resources/lb_v2`

Summary of the Pull Request
Use client from context in

resource/opentelekomcloud_lb_member_v2
resource/opentelekomcloud_lb_monitor_v2
resource/opentelekomcloud_lb_pool_v2
resource/opentelekomcloud_lb_whitelist_v2
resource/opentelekomcloud_lb_loadbalancer_v2
resource/opentelekomcloud_lb_listener_v2
resource/opentelekomcloud_lb_l7rule_v2
resource/opentelekomcloud_lb_l7policy_v2
resource/opentelekomcloud_lb_certificate_v2

PR Checklist

 Refers to: #1848
 Tests added/passed.
 Release notes added.

Acceptance Steps Performed
=== RUN   TestAccLBV2Whitelist_basic
=== PAUSE TestAccLBV2Whitelist_basic
=== CONT  TestAccLBV2Whitelist_basic
--- PASS: TestAccLBV2Whitelist_basic (137.18s)
PASS


Process finished with the exit code 0

=== RUN   TestAccLBV2Pool_basic
=== PAUSE TestAccLBV2Pool_basic
=== CONT  TestAccLBV2Pool_basic
--- PASS: TestAccLBV2Pool_basic (168.53s)
=== RUN   TestAccLBV2Pool_persistenceNull
=== PAUSE TestAccLBV2Pool_persistenceNull
=== CONT  TestAccLBV2Pool_persistenceNull
--- PASS: TestAccLBV2Pool_persistenceNull (113.58s)
PASS


Process finished with the exit code 0

=== RUN   TestAccLBV2Monitor_basic
=== PAUSE TestAccLBV2Monitor_basic
=== CONT  TestAccLBV2Monitor_basic
--- PASS: TestAccLBV2Monitor_basic (190.35s)
=== RUN   TestAccLBV2Monitor_minConfig
=== PAUSE TestAccLBV2Monitor_minConfig
=== CONT  TestAccLBV2Monitor_minConfig
--- PASS: TestAccLBV2Monitor_minConfig (190.90s)
PASS

Process finished with the exit code 0

=== RUN   TestAccLBV2Member_basic
=== PAUSE TestAccLBV2Member_basic
=== CONT  TestAccLBV2Member_basic
--- PASS: TestAccLBV2Member_basic (231.74s)
=== RUN   TestAccLBV2Member_import
=== PAUSE TestAccLBV2Member_import
=== CONT  TestAccLBV2Member_import
--- PASS: TestAccLBV2Member_import (154.02s)
PASS


Process finished with the exit code 0

=== RUN   TestAccLBV2LoadBalancer_basic
=== PAUSE TestAccLBV2LoadBalancer_basic
=== CONT  TestAccLBV2LoadBalancer_basic
--- PASS: TestAccLBV2LoadBalancer_basic (112.45s)
=== RUN   TestAccLBV2LoadBalancer_import
=== PAUSE TestAccLBV2LoadBalancer_import
=== CONT  TestAccLBV2LoadBalancer_import
--- PASS: TestAccLBV2LoadBalancer_import (72.07s)
PASS


Process finished with the exit code 0

=== RUN   TestAccLBV2L7Rule_basic
=== PAUSE TestAccLBV2L7Rule_basic
=== CONT  TestAccLBV2L7Rule_basic
--- PASS: TestAccLBV2L7Rule_basic (179.52s)
PASS


Process finished with the exit code 0

=== RUN   TestAccLBV2L7Policy_basic
=== PAUSE TestAccLBV2L7Policy_basic
=== CONT  TestAccLBV2L7Policy_basic
--- PASS: TestAccLBV2L7Policy_basic (118.36s)
PASS


Process finished with the exit code 0

=== RUN   TestAccLBV2Certificate_basic
=== PAUSE TestAccLBV2Certificate_basic
=== CONT  TestAccLBV2Certificate_basic
--- PASS: TestAccLBV2Certificate_basic (135.80s)
=== RUN   TestAccLBV2Certificate_import
=== PAUSE TestAccLBV2Certificate_import
=== CONT  TestAccLBV2Certificate_import
--- PASS: TestAccLBV2Certificate_import (51.91s)
PASS


Process finished with the exit code 0

=== RUN   TestAccLBV2Listener_tls
=== PAUSE TestAccLBV2Listener_tls
=== CONT  TestAccLBV2Listener_tls
--- PASS: TestAccLBV2Listener_tls (144.58s)
PASS


Process finished with the exit code 0

Reviewed-by: Aloento <None>
Reviewed-by: Vladimir Vshivkov <None>

* [VPC] Use client from ctx in `vpc_2` (#1853)

[VPC] Use client from ctx in `vpc_2`

Summary of the Pull Request
Use client from ctx in floatingip_v2, foatingip_associate_v2, vpc_peering_connection_v2, port_v2, vpc_route_v2
PR Checklist

 Refers to: #1848
 Tests added/passed.
 Schema updated.
 Release notes added.

Acceptance Steps Performed
Read comments

Reviewed-by: Artem Lifshits <None>
Reviewed-by: Anton Sidelnikov <None>

* [IAM]: email validation added & resource Update fix (#1856)

[IAM]: email validation added & resource Update fix

Summary of the Pull Request
This PR adds validation for email schema.
PR Checklist

 Refers to:  #1850
 Closes: #1855
 Tests added/passed.
 Schema updated.
 Release notes added.

Acceptance Steps Performed
=== RUN   TestAccIdentityV3User_basic
--- PASS: TestAccIdentityV3User_basic (68.59s)
=== RUN   TestAccIdentityV3User_importBasic
--- PASS: TestAccIdentityV3User_importBasic (64.83s)
PASS

Process finished with the exit code 0

Reviewed-by: Rodion Gyrbu <fpsoff@outlook.com>
Reviewed-by: Aloento <None>
Reviewed-by: Anton Sidelnikov <None>
Reviewed-by: Artem Lifshits <None>

Co-authored-by: Rodion Gyrbu <rodion.gyrbu@t-systems.com>
Co-authored-by: Rodion Gyrbu <rodion.gyrbu@nexign.com>
Co-authored-by: Artem Lifshits <55093318+artem-lifshits@users.noreply.github.com>

Aloento committed:
Merge remote-tracking branch 'origin/devel' into capacity

Aloento commented:
@artem-lifshits @anton-sidelnikov could you run the test again? Thanks.

Aloento committed:
check 0

Aloento committed:
Merge remote-tracking branch 'origin/devel' into capacity

vladimirvshivkov commented on releasenotes/notes/change-capacity-to-float32-2ac71ac93c328888.yaml:
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    **[DCS]** Change Capacity from int to float32 (`#1842 <https://github.com/opentelekomcloud/terraform-provider-opentelekomcloud/issues/1842>`_)
```
float64

vladimirvshivkov commented:
```
=== RUN   TestAccDcsAZV1DataSource_basic
=== PAUSE TestAccDcsAZV1DataSource_basic
=== CONT  TestAccDcsAZV1DataSource_basic
--- PASS: TestAccDcsAZV1DataSource_basic (34.52s)
=== RUN   TestAccDcsMaintainWindowV1DataSource_basic
=== PAUSE TestAccDcsMaintainWindowV1DataSource_basic
=== CONT  TestAccDcsMaintainWindowV1DataSource_basic
--- PASS: TestAccDcsMaintainWindowV1DataSource_basic (34.45s)
=== RUN   TestAccDcsProductV1DataSource_basic
=== PAUSE TestAccDcsProductV1DataSource_basic
=== CONT  TestAccDcsProductV1DataSource_basic
--- PASS: TestAccDcsProductV1DataSource_basic (35.24s)
=== RUN   TestAccDcsInstancesV1_basic
=== PAUSE TestAccDcsInstancesV1_basic
=== CONT  TestAccDcsInstancesV1_basic
--- PASS: TestAccDcsInstancesV1_basic (131.72s)
=== RUN   TestAccDcsInstancesV1_basicSingleInstance
=== PAUSE TestAccDcsInstancesV1_basicSingleInstance
=== CONT  TestAccDcsInstancesV1_basicSingleInstance
--- PASS: TestAccDcsInstancesV1_basicSingleInstance (73.84s)
PASS

Process finished with the exit code 0
```

Aloento committed:
Update change-capacity-to-float32-2ac71ac93c328888.yaml
